lang/ruby/lib/avro/schema.rb

```diff
@@ -33,7 +35,10 @@ class Schema
     LONG_MAX_VALUE = (1 << 63) - 1
 
     def self.parse(json_string)
-      real_parse(MultiJson.load(json_string), {})
+      json_obj = MultiJson.load(json_string)
+      schema = real_parse(json_obj, {})
+      schema.logical_type = LogicalTypes.parse(json_obj) if json_obj.is_a?(Hash)
```

Comment:
I see what you're saying about the difficulty of doing it that way, which is pretty annoying. But, I think it would be better to avoid exposing the accessor.